mobile/src/main/java/ch/epfl/sdp/mobile/application/chess/implementation/PersistentGame.kt

```diff
@@ -12,15 +12,29 @@ import ch.epfl.sdp.mobile.application.chess.rules.Role
  * @param nextPlayer the [Color] of the next player to play.
  */
 data class PersistentGame(
-    override val previous: Pair<Game, Action>?,
+    override val previous: Pair<PersistentGame, Action>?,
     val nextPlayer: Color,
     override val board: Board<Piece<Color>>,
 ) : Game {
 
+  /**
+   * An implementation of a [BoardWithHistory] which delegates retrieving information to the current
+   * [board] or the [previous] [PersistentGame].
+   *
+   * @param player the [Color] for which the board should be normalized.
+   */
+  private inner class PersistentNormalizedGameBoardWithHistory(
+      private val player: Color,
+  ) : BoardWithHistory<Piece<Role>>, Board<Piece<Role>> by NormalizedBoardDecorator(player, board) {
+
+    override val previousBoardWithHistory: BoardWithHistory<Piece<Role>>?
+      get() = previous?.let { (game, _) -> game.PersistentNormalizedGameBoardWithHistory(player) }
```

Comment:
[Question] Why sometimes we chose to override get() explicitly and other times we simply drop it ?